src/lib/interp.rs

```diff
@@ -126,9 +133,40 @@ impl VM {
                 }
                 Instr::LOAD_GLOBAL(ref name) => panic!("NotYetImplemented"),
                 Instr::STORE_GLOBAL(ref name) => panic!("NotYetImplemented"),
-                Instr::NEW_OBJECT(ref class_name) => panic!("NotYetImplemented"),
-                Instr::LOAD_FIELD(ref field_name) => panic!("NotYetImplemented"),
-                Instr::STORE_FIELD(ref field_name) => panic!("NotYetImplemented"),
+                Instr::NEW_OBJECT(ref class_name) => {
+                    let obj = Object::new(class_name.to_string());
+                    self.heap.push(obj);
+                    let obj_ref = self.heap.len() - 1;
+                    let frame = self.frames.last_mut().unwrap();
+                    frame.push(NativeType::ObjectRef(obj_ref));
+                    self.pc += 1
+                },
+                Instr::LOAD_FIELD(ref field_name) => {
+                    let frame = self.frames.last_mut().unwrap();
```

Comment:
Wouldn't it be easier to have a function on all objects called `get_attr` (or `load_field` -- though I think that's a slightly misleading name myself!)? Those objects that can't provide this functionality can then panic.

Reply:
I think so yeah, but I think this should be an improvement made in a future PR where classes and class inheritance is used correctly, this way I can refactor to make method calls attributes too and fall back on method calls if fields are not found.